js/server/tests/aql/aql-queries-simple.js

```diff
@@ -54,6 +54,16 @@ function ahuacatlQuerySimpleTestSuite () {
 
     tearDown : function () {
     },
+    
+    testNoArraySorting1 : function () {
+      let query = "LET values = [9,16,8,15,7,14,6,13,5,12,4,11,3,10,2,1] RETURN values";
+      assertEqual([9, 16, 8, 15, 7, 14, 6, 13, 5, 12, 4, 11, 3, 10, 2, 1], AQL_EXECUTE(query).json[0]);
+    },
+    
+    testNoArraySorting2 : function () {
+      let query = "LET values = [9,16,8,15,7,14,6,13,5,12,4,11,3,10,2,1] FOR v IN values FILTER v IN values LIMIT 1 RETURN values";
+      assertEqual([9, 16, 8, 15, 7, 14, 6, 13, 5, 12, 4, 11, 3, 10, 2, 1], AQL_EXECUTE(query).json[0]);
```

Comment:
I assume you've confirmed that these added tests fail prior to the change?

I wasn't able to reduce the test case to something this brief but it may have been because the bug only surfaced with arrays with more than 16 items.

Otherwise LGTM.

Reply:
Yes, one of the two new tests fails in vanilla 3.3, and is fixed by this PR.